src/schema/meta/expression_tests.yaml

```diff
@@ -77,15 +84,61 @@
   result: false
 - expression: exists([], null)
   result: false
+
+# Truth/falsity of final expressions
 - expression: evaluate(true)
   result: true
 - expression: evaluate(false)
   result: false
 - expression: evaluate(null)
   result: false
+
+# General expressions
 - expression: 1 + 2
   result: 3
 - expression: '"cat" + "dog"'
   result: 'catdog'
 - expression: '1 + "cat"'
   result: null
+- expression: match('string', '.*')
+  result: true
+- expression: match('', '.')
+  result: false
+- expression: substr('string', 1, 4)
+  result: 'tri'
+- expression: substr('string', 0, 20)
+  result: 'string'
+- expression: type(1)
+  result: 'number'
+- expression: type([])
+  result: 'array'
+- expression: type({})
+  result: 'object'
+- expression: type(true)
+  result: 'boolean'
+- expression: intersects([1], [1, 2])
+  result: true
+- expression: intersects([1], [])
+  result: false
+- expression: length([1, 2, 3])
+  result: 3
+- expression: length([])
+  result: 0
+- expression: count([1, 2, 3], 1)
+  result: 1
+- expression: index(["i", "j", "k"], "i")
+  result: 0
+- expression: index(["i", "j", "k"], "j")
+  result: 1
+- expression: index(["i", "j", "k"], "x")
+  result: null
+- expression: sorted([3, 2, 1])
+  result: [1, 2, 3]
```

Comment:
Are there any non-numeric cases for sorted in the spec?

Reply:
The only use case we've found is for checking that certain arrays are monotonically increasing.

Reply:
awesome!